src/client/gpm_import_and_canon_name.c

```diff
@@ -261,6 +269,14 @@ OM_uint32 gpm_canonicalize_name(OM_uint32 *minor_status,
         goto done;
     }
 
+    if (localname) {
+        ret_min = gp_add_option(&arg->options.options_val,
```

Comment:
This isn't checked and is reassigned on line 286.  Probably we should skip assignment for clarity, unless you want to bail out on failure here.